### DISCUSSION TITLE
Maintaining multiple similar repos - best practice

### DISCUSSION THREAD

**Discussion by do-me** [squidfunk/mkdocs-material#4901] [General]
Hi everyone, 

I'd like to ask some general advice here but I guess it has the potential to turn into a longer opiniated discussion. 

### Situation
Our company has a product suite of **5 products.** Each product has it's own mkdocs-material wiki. The general structure of the wikis (e.g. customized landing page & icons), the used plugins, settings etc. is identical for each wiki but there is different content.

### Problem
Maintaining is on the edge of becoming a pain as I need to update the source code of all wikis individually. 

### Question
What are the options for simplifying this workflow? 

### Thoughts 
I have a couple of solution ideas: 

0. (Keep it as is until it becomes an _actual_ overhead) 
1. Using git submodule. So basically putting everything the wikis have in common in one submodule. Did anyone ever try with mkdocs-material?
2.  Using some kind of third-party app. I imagine it to work like this: reference all repos and all files to be updated in the repos. Tell the app _where_ to insert/update/delete a line and push the changes. Only the _where_ part seems to be difficult if the config files have a slightly different structure. I'd prefer a CLI but maybe someone developed a GUI for this before? Anyway, this approach seems to have the potential to mess up all repos in the blink of an eye.

**Comment by squidfunk** [comment]
https://github.com/squidfunk/mkdocs-material/discussions/4226#discussioncomment-4748913 explains how to package a preconfigured version of Material for MkDocs. This will work if your sites are (almost) the same. If you deviate, it might be a better idea to create [a plugin or hook which preconfigures the site](https://github.com/squidfunk/mkdocs-material/pull/4847#issuecomment-1378467860) according to your needs.
